tests/mock_clients/mock_gs.py

```diff
@@ -113,30 +113,49 @@ def get_blob(self, blob):
         else:
             return None
 
-    def list_blobs(self, max_results=None, prefix=None):
+    def list_blobs(self, max_results=None, prefix=None, delimiter=None):
         path = self.name if prefix is None else self.name / prefix
-        items = [
-            MockBlob(self.name, f.relative_to(self.name), client=self.client)
-            for f in path.glob("**/*")
-            if f.is_file() and not f.name.startswith(".")
-        ]
+        files = [f for f in path.glob("**/*") if f.is_file() and not f.name.startswith(".")]
+        sub_directories = [str(f.relative_to(self.name)) for f in path.glob("*") if f.is_dir()]
 
+        # filter blobs by delimiter
+        if delimiter == "/":
+            files = [file for file in files if len(file.relative_to(path).parents) == 1]
+
+        blobs = [MockBlob(self.name, f.relative_to(self.name), client=self.client) for f in files]
         # bucket name for passing tests
         if self.bucket_name == DEFAULT_GS_BUCKET_NAME:
-            return iter(MockHTTPIterator(items, max_results))
+            return MockHTTPIterator(blobs, sub_directories, max_results)
```

Comment:
I think it will be a little easer to read and maintain to not recursively glob if `delimiter` is passed. E.g., something like the follow (untested) code

```python
    def list_blobs(self, max_results=None, prefix=None, delimiter=None):
        path = self.name if prefix is None else self.name / prefix
        pattern = "**/*" if delimiter is None else "*"
        blobs, prefixes = [], []
        for item in path.glob(pattern):
            if not item.name.startswith("."):
                if item.is_file():
                    blobs.append(MockBlob(self.name, f.relative_to(self.name), client=self.client))
                else:
                    prefixes.append(str(f.relative_to(self.name).as_posix()))

        # bucket name for passing tests
        if self.bucket_name == DEFAULT_GS_BUCKET_NAME:
            return MockHTTPIterator(blobs, prefixes, max_results)
```